packages/teams-js/src/private/externalAppAuthenticationForCEA.ts

```diff
@@ -0,0 +1,269 @@
+import { sendMessageToParentAsync } from '../internal/communication';
+import { ensureInitialized } from '../internal/internalAPIs';
+import { ApiName, ApiVersionNumber, getApiVersionTag } from '../internal/telemetry';
+import { validateId } from '../internal/utils';
+import { AppId } from '../public';
+import { errorNotSupportedOnPlatform, FrameContexts } from '../public/constants';
+import { runtime } from '../public/runtime';
+import { externalAppAuthentication } from './externalAppAuthentication';
+
+const externalAppAuthenticationTelemetryVersionNumber: ApiVersionNumber = ApiVersionNumber.V_2;
+
+/**
+ * @hidden
+ * Namespace to delegate authentication requests to the host for custom engine agents
+ * @internal
+ * Limited to Microsoft-internal use
+ * @beta
+ */
+export namespace externalAppAuthenticationForCEA {
+  /**
+   * @beta
+   * @hidden
+   * Signals to the host to perform SSO authentication for the application specified by the app ID, and then send the authResult to the application backend.
+   * @internal
+   * Limited to Microsoft-internal use
+   * @param appId Id of the application backend for which the host should attempt SSO authentication.
+   * @param conversationId ConversationId To tell the bot what conversation the calls are coming from
+   * @param authTokenRequest Parameters for SSO authentication
+   * @throws InvokeError if the host encounters an error while authenticating
+   * @returns A promise that resolves when authentication succeeds and rejects with InvokeError on failure
+   */
+  export async function authenticateWithSSO(
+    appId: AppId,
+    conversationId: string,
+    authTokenRequest: externalAppAuthentication.AuthTokenRequestParameters,
+  ): Promise<void> {
+    ensureInitialized(runtime, FrameContexts.content);
+
+    if (!isSupported()) {
+      throw errorNotSupportedOnPlatform;
+    }
+
+    validateId(conversationId, new Error('conversation id is not valid.'));
+
+    const [error] = await sendMessageToParentAsync<[externalAppAuthentication.InvokeError]>(
```

Comment:
>[externalAppAuthentication.InvokeError [](http://example.com/codeflow?start=51&length=38)

It seems unnecessary to have a single value in an array; can you use something like `sendAndUnwrap` instead of `sendMessageToParentAsync` so that you just get back the value you want and not an array with the value in it?